sloglint.go

```diff
@@ -113,6 +116,14 @@ func run(pass *analysis.Pass, opts *Options) {
 			return
 		}
 
+		if opts.ContextOnly {
+			fnName := fn.FullName()
+
+			if !strings.HasSuffix(fnName, ".Log") && !strings.HasSuffix(fnName, "Context") {
```

Comment:
WDYT if we check whether the first element of the `call.Args` slice is a context instead? Then no `.Log` special handling would be needed 🤔

Reply:
The thought did cross my mind as I was writing it originally.

I've wwitched to checking the type of first arg, and added a couple of variations to the tests to ensure identifiers (`ctx`) and call expressions (`context.Background()`, etc.) behave the same. I think the test cover all the bases, but let me know if you can think of any other edge cases to prove out.

Reply:
Not sure we need to cover different context cases (`context.Background()`, `context.TODO()`, etc), since we only check the type, and it is always the same.